lib/Listener/OutboxStatusChangeListener.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright 2024 Anna Larch <anna.larch@gmx.net>
+ *
+ * @author Anna Larch <anna.larch@gmx.net>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Mail\Listener;
+
+use OCA\Mail\Db\LocalMessageMapper;
+use OCA\Mail\Events\OutboxMessageStatusChangeEvent;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+
+/**
+ * @template-implements IEventListener<Event|OutboxMessageStatusChangeEvent>
+ */
+class OutboxStatusChangeListener implements IEventListener {
+
+	public function __construct(private LocalMessageMapper $messageMapper) {
+	}
+
+	public function handle(Event $event): void {
+		if (!($event instanceof OutboxMessageStatusChangeEvent)) {
+			return;
+		}
+
+		$message = $event->getMessage();
+		$this->messageMapper->update($message);
+
+	}
+}
```

Comment:
this does not feel like a use case for events. can you inline the mapper call and make persistence explicit?

Reply:
yeah that event can go, it's not needed with the handlers any more.